packages/teuchos/core/src/Teuchos_ArrayRCP.hpp

```diff
@@ -164,7 +164,7 @@ ArrayRCP<T>::ArrayRCP(
     }
     if (existing_RCPNode) {
       // Will not call add_new_RCPNode(...)
-      node_ = RCPNodeHandle(existing_RCPNode, RCP_WEAK, false);
+      node_ = RCPNodeHandle(existing_RCPNode, RCP_WEAK, RCP_STRONG, false); // source is strong because we are constructing from a ptr
```

Comment:
Okay, if you need another argument for the RCPNodeHandle() constructor that is okay but is this really needed?